src/magicproxy/async_proxy.py

```diff
@@ -110,12 +113,35 @@ def _clean_response_headers(headers):
             f"Disallowed by GitHub proxy. Allowed scopes: {', '.join(token_info.scopes)}"
         )
 
+    path = _clean_path_queries(path)
+
     return await _proxy_request(
         request=request,
         url=f"{GITHUB_API_ROOT}/{path}",
         headers={"Authorization": f"Bearer {token_info.github_token}"},
     )
 
+def _clean_path_queries(path) -> str:
+    for param in _query_params_to_clean:
+        path, replaced = re.subn('({}=[\w*-+.%]*)(?:$|\&)'.format(param),'',path)
```

Comment:
Use urlsplit and parse_qs to split and parse the string, then urljoin to put it back together instead of this regex.

Reply:
Will do :+1: